tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ManagedIdentityTests.cs

```
@@ -0,0 +1,199 @@
+        }
+
+        [DataTestMethod]
+        [DataRow(MsiAzureResource.WebApp, "", DisplayName = "System Identity Web App")]
+        [DataRow(MsiAzureResource.Function, "", DisplayName = "System Identity Function App")]
+        [DataRow(MsiAzureResource.WebApp, UserAssignedClientID, DisplayName = "User Identity Web App")]
+        [DataRow(MsiAzureResource.Function, UserAssignedClientID, DisplayName = "User Identity Function App")]
+        [DataRow(MsiAzureResource.WebApp, Mi_res_id, DisplayName = "ResourceID Web App")]
+        [DataRow(MsiAzureResource.Function, Mi_res_id, DisplayName = "ResourceID Function App")]
+        public async Task AcquireMSITokenAsync(MsiAzureResource azureResource, string userIdentity)
+        {
+            //Arrange
+            AuthenticationResult result = null;
+
+            using (new EnvVariableContext())
+            {
+                //Get the Environment Variables
+                Dictionary<string, string> envVariables = 
+                    await GetEnvironmentVariablesAsync(azureResource).ConfigureAwait(false);
+
+                //Set the Environment Variables
+                SetEnvironmentVariables(envVariables);
+
+                //form the http proxy URI 
+                string uri = s_baseURL + $"GetMSIToken?" +
+                    $"azureresource={azureResource.ToString().ToLowerInvariant()}&uri=";
+
+                //Create CCA with Proxy
+                IConfidentialClientApplication cca = CreateCCAWithProxy(uri);
+
+                //Act
+                try
+                {
+                    result = await cca
+                        .AcquireTokenForClient(s_msi_scopes)
+                        .WithManagedIdentity(userIdentity)
+                        .ExecuteAsync().ConfigureAwait(false);
+                }
+                catch (MsalClientException ex)
+                {
+                    throw new Exception(ex.Message);
+                }
```

Not needed? Just let it bubble up and break the test?